```
Run export PATH=${HOME}/Library/Python/3.8/bin:${HOME}/Library/Python/3.9/bin:${HOME}/.local/bin:${PATH}
************* Module version
sys/version.py:15:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module syscall_preprocessing
sys/syscall_preprocessing.py:11:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
sys/syscall_preprocessing.py:12:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module create_tags_rz
sys/create_tags_rz.py:12:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module check_meson_subproject
sys/check_meson_subproject.py:21:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
sys/check_meson_subproject.py:46:21: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module cmd_descs_generate
librz/core/cmd_descs/cmd_descs_generate.py:822:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
librz/core/cmd_descs/cmd_descs_generate.py:825:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
librz/core/cmd_descs/cmd_descs_generate.py:836:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
librz/core/cmd_descs/cmd_descs_generate.py:839:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
************* Module openbsd-gen
test/unit/legacy_unit/syscall/openbsd-gen.py:8:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
test/unit/legacy_unit/syscall/openbsd-gen.py:40:5: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```

--- a/librz/core/cmd_api.c
+++ b/librz/core/cmd_api.c
@@ -79,6 +79,16 @@ RZ_IPI int rz_output_mode_to_char(RzOutputMode mode) {
 	return -1;
 }
 
+RZ_IPI const char *rz_output_mode_to_summary(RzOutputMode mode) {
+	size_t i;
+	for (i = 0; i < RZ_ARRAY_SIZE(argv_modes); i++) {
+		if (argv_modes[i].mode == mode) {
+			return argv_modes[i].summary_suffix;
+		}
+	}
+	return "";
+}
+
 static int value = 0;
 
 #define NCMDS (sizeof(cmd->cmds) / sizeof(*cmd->cmds))
@@ -330,6 +340,42 @@ out:
 	return res;
 }
 
+/**
+ * \brief Set the default mode of the command descriptor, if the type allows it.
+ *
+ * Command descriptors that support multiple output modes can also have a
+ * default one. This function can be used to set it.
+ *
+ * \return True if the default output mode was changed, false otherwise.
+ */
+RZ_API bool rz_cmd_desc_set_default_mode(RzCmdDesc *cd, RzOutputMode mode) {
+	rz_return_val_if_fail(cd, false);
+
+	switch (cd->type) {
+	case RZ_CMD_DESC_TYPE_ARGV_MODES:
+		if (cd->d.argv_modes_data.modes & RZ_OUTPUT_MODE_STANDARD) {
+			return false;
+		}
+		cd->d.argv_modes_data.default_mode = mode;
+		return true;
+	case RZ_CMD_DESC_TYPE_ARGV_STATE:
+		if (cd->d.argv_state_data.modes & RZ_OUTPUT_MODE_STANDARD) {
+			return false;
+		}
+		cd->d.argv_state_data.default_mode = mode;
+		return true;
+	case RZ_CMD_DESC_TYPE_GROUP: {
+		RzCmdDesc *exec_cd = rz_cmd_desc_get_exec(cd);
+		if (exec_cd) {
+			return rz_cmd_desc_set_default_mode(exec_cd, mode);
+		}
+		return false;
+	}
+	default:
+		return false;
+	}
+}
+
 RZ_API char **rz_cmd_alias_keys(RzCmd *cmd, int *sz) {
 	if (sz) {
 		*sz = cmd->aliases.count;
@@ -543,11 +589,30 @@ static void get_minmax_argc(RzCmdDesc *cd, int *min_argc, int *max_argc) {
 	}
 }
 
+static RzOutputMode get_cd_default_mode(RzCmdDesc *cd) {
+	switch (cd->type) {
+	case RZ_CMD_DESC_TYPE_ARGV_MODES:
+		return cd->d.argv_modes_data.default_mode;
+	case RZ_CMD_DESC_TYPE_ARGV_STATE:
+		return cd->d.argv_state_data.default_mode;
+	default:
+		return RZ_OUTPUT_MODE_STANDARD;
+	}
+}
+
+static bool has_cd_default_mode(RzCmdDesc *cd) {
+	return get_cd_default_mode(cd) != RZ_OUTPUT_MODE_STANDARD;
+}
+
 static RzOutputMode cd_suffix2mode(RzCmdDesc *cd, const char *cmdid) {
 	if (!has_cd_submodes(cd)) {
 		return 0;
 	}
-	return suffix2mode(cmdid + strlen(cd->name));
+	RzOutputMode mode = suffix2mode(cmdid + strlen(cd->name));
+	if (mode == RZ_OUTPUT_MODE_STANDARD && has_cd_default_mode(cd)) {
+		mode = get_cd_default_mode(cd);
+	}
+	return mode;
 }
 
 /**
@@ -897,10 +962,17 @@ static void fill_usage_strbuf(RzCmd *cmd, RzStrBuf *sb, RzCmdDesc *cd, bool use_
 		rz_strbuf_append(sb, pal_reset);
 	}
 	if (cd->help->summary) {
-		columns += strbuf_append_calc(sb, "   ");
-		rz_strbuf_append(sb, pal_help_color);
-		fill_wrapped_comment(cmd, sb, cd->help->summary, columns);
-		rz_strbuf_append(sb, pal_reset);
+		RzStrBuf *summary_sb = rz_strbuf_new(cd->help->summary);
+		if (summary_sb) {
+			columns += strbuf_append_calc(sb, "   ");
+			rz_strbuf_append(sb, pal_help_color);
+			if (has_cd_default_mode(cd)) {
+				rz_strbuf_appendf(summary_sb, "%s", rz_output_mode_to_summary(get_cd_default_mode(cd)));
+			}
+			fill_wrapped_comment(cmd, sb, rz_strbuf_get(summary_sb), columns);
+			rz_strbuf_append(sb, pal_reset);
+			rz_strbuf_free(summary_sb);
+		}
 	}
 	rz_strbuf_append(sb, "\n");
 }
@@ -1953,6 +2025,7 @@ static RzCmdDesc *argv_modes_new(RzCmd *cmd, RzCmdDesc *parent, const char *name
 
 	res->d.argv_modes_data.cb = cb;
 	res->d.argv_modes_data.modes = modes;
+	res->d.argv_modes_data.default_mode = RZ_OUTPUT_MODE_STANDARD;
 	get_minmax_argc(res, &res->d.argv_modes_data.min_argc, &res->d.argv_modes_data.max_argc);
 	return res;
 }
@@ -1965,6 +2038,7 @@ static RzCmdDesc *argv_state_new(RzCmd *cmd, RzCmdDesc *parent, const char *name
 
 	res->d.argv_state_data.cb = cb;
 	res->d.argv_state_data.modes = modes;
+	res->d.argv_state_data.default_mode = RZ_OUTPUT_MODE_STANDARD;
 	get_minmax_argc(res, &res->d.argv_state_data.min_argc, &res->d.argv_state_data.max_argc);
 	return res;
 }

--- a/librz/core/cmd_descs/cmd_descs.yaml
+++ b/librz/core/cmd_descs/cmd_descs.yaml
@@ -22,6 +22,8 @@
 #   modes: >
 #     same as RzCmdDesc.modes. When present the type is automatically set to
 #     ARGV_MODES.
+#   default_mode: >
+#     same as RzCmdDesc.default_mode.
 #   details: >
 #     an array of RzCmdDescDetail or a string referencing an already existing
 #     RzCmdDescDetail array

--- a/librz/core/cmd_descs/cmd_descs_generate.py
+++ b/librz/core/cmd_descs/cmd_descs_generate.py
@@ -117,6 +117,9 @@ DEFINE_FAKE_TEMPLATE = """
 \tRzCmdDesc *{cname}_cd = rz_cmd_desc_fake_new(core->rcmd, {parent_cname}_cd, {name}, &{help_cname});
 \trz_warn_if_fail({cname}_cd);"""
 
+SET_DEFAULT_MODE_TEMPLATE = """
+\trz_cmd_desc_set_default_mode({cname}_cd, {default_mode});"""
+
 CD_TYPE_OLDINPUT = "RZ_CMD_DESC_TYPE_OLDINPUT"
 CD_TYPE_GROUP = "RZ_CMD_DESC_TYPE_GROUP"
 CD_TYPE_ARGV = "RZ_CMD_DESC_TYPE_ARGV"
@@ -429,6 +432,7 @@ class CmdDesc:
         self.exec_cd = None
         self.modes = c.pop("modes", None)
         self.handler = c.pop("handler", None)
+        self.default_mode = c.pop("default_mode", None)
         # RzCmdDescHelp fields
         self.summary = strip(c.pop("summary"))
         self.description = strip(c.pop("description", None))
@@ -621,6 +625,60 @@ class CmdDesc:
         return self.str_tab()
 
 
+def createcd_typegroup(cd):
+    if cd.exec_cd and cd.exec_cd.type == CD_TYPE_ARGV_MODES:
+        formatted_string = DEFINE_GROUP_MODES_TEMPLATE.format(
+            cname=cd.cname,
+            parent_cname=cd.parent.cname,
+            name=strornull(cd.name),
+            modes=" | ".join(cd.exec_cd.modes),
+            handler_cname=cd.exec_cd.get_handler_cname(),
+            help_cname_ref="&" + cd.exec_cd.get_help_cname(),
+            group_help_cname=cd.get_help_cname(),
+        )
+        if cd.exec_cd.default_mode is not None:
+            formatted_string += SET_DEFAULT_MODE_TEMPLATE.format(
+                cname=cd.cname,
+                default_mode=cd.exec_cd.default_mode,
+            )
+        formatted_string += "\n".join(
+            [createcd(child) for child in cd.subcommands[1:] or []]
+        )
+    elif cd.exec_cd and cd.exec_cd.type == CD_TYPE_ARGV_STATE:
+        formatted_string = DEFINE_GROUP_STATE_TEMPLATE.format(
+            cname=cd.cname,
+            parent_cname=cd.parent.cname,
+            name=strornull(cd.name),
+            modes=" | ".join(cd.exec_cd.modes),
+            handler_cname=cd.exec_cd.get_handler_cname(),
+            help_cname_ref="&" + cd.exec_cd.get_help_cname(),
+            group_help_cname=cd.get_help_cname(),
+        )
+        if cd.exec_cd.default_mode is not None:
+            formatted_string += SET_DEFAULT_MODE_TEMPLATE.format(
+                cname=cd.cname,
+                default_mode=cd.exec_cd.default_mode,
+            )
+        formatted_string += "\n".join(
+            [createcd(child) for child in cd.subcommands[1:] or []]
+        )
+    else:
+        formatted_string = DEFINE_GROUP_TEMPLATE.format(
+            cname=cd.cname,
+            parent_cname=cd.parent.cname,
+            name=strornull(cd.name),
+            handler_cname=(cd.exec_cd and cd.exec_cd.get_handler_cname()) or "NULL",
+            help_cname_ref=(cd.exec_cd and "&" + cd.exec_cd.get_help_cname()) or "NULL",
+            group_help_cname=cd.get_help_cname(),
+        )
+        subcommands = (
+            cd.exec_cd and cd.subcommands and cd.subcommands[1:]
+        ) or cd.subcommands
+        formatted_string += "\n".join([createcd(child) for child in subcommands or []])
+
+    return formatted_string
+
+
 def createcd(cd):
     formatted_string = None
 
@@ -641,6 +699,11 @@ def createcd(cd):
             handler_cname=cd.get_handler_cname(),
             help_cname=cd.get_help_cname(),
         )
+        if cd.default_mode is not None:
+            formatted_string += SET_DEFAULT_MODE_TEMPLATE.format(
+                cname=cd.cname,
+                default_mode=cd.default_mode,
+            )
     elif cd.type == CD_TYPE_ARGV_STATE:
         formatted_string = DEFINE_ARGV_STATE_TEMPLATE.format(
             cname=cd.cname,
@@ -650,6 +713,11 @@ def createcd(cd):
             handler_cname=cd.get_handler_cname(),
             help_cname=cd.get_help_cname(),
         )
+        if cd.default_mode is not None:
+            formatted_string += SET_DEFAULT_MODE_TEMPLATE.format(
+                cname=cd.cname,
+                default_mode=cd.default_mode,
+            )
     elif cd.type == CD_TYPE_FAKE:
         formatted_string = DEFINE_FAKE_TEMPLATE.format(
             cname=cd.cname,
@@ -678,53 +746,8 @@ def createcd(cd):
         formatted_string += "\n".join(
             [createcd(child) for child in cd.subcommands or []]
         )
-    elif (
-        cd.type == CD_TYPE_GROUP
-        and cd.exec_cd
-        and cd.exec_cd.type == CD_TYPE_ARGV_MODES
-    ):
-        formatted_string = DEFINE_GROUP_MODES_TEMPLATE.format(
-            cname=cd.cname,
-            parent_cname=cd.parent.cname,
-            name=strornull(cd.name),
-            modes=" | ".join(cd.exec_cd.modes),
-            handler_cname=cd.exec_cd.get_handler_cname(),
-            help_cname_ref="&" + cd.exec_cd.get_help_cname(),
-            group_help_cname=cd.get_help_cname(),
-        )
-        formatted_string += "\n".join(
-            [createcd(child) for child in cd.subcommands[1:] or []]
-        )
-    elif (
-        cd.type == CD_TYPE_GROUP
-        and cd.exec_cd
-        and cd.exec_cd.type == CD_TYPE_ARGV_STATE
-    ):
-        formatted_string = DEFINE_GROUP_STATE_TEMPLATE.format(
-            cname=cd.cname,
-            parent_cname=cd.parent.cname,
-            name=strornull(cd.name),
-            modes=" | ".join(cd.exec_cd.modes),
-            handler_cname=cd.exec_cd.get_handler_cname(),
-            help_cname_ref="&" + cd.exec_cd.get_help_cname(),
-            group_help_cname=cd.get_help_cname(),
-        )
-        formatted_string += "\n".join(
-            [createcd(child) for child in cd.subcommands[1:] or []]
-        )
     elif cd.type == CD_TYPE_GROUP:
-        formatted_string = DEFINE_GROUP_TEMPLATE.format(
-            cname=cd.cname,
-            parent_cname=cd.parent.cname,
-            name=strornull(cd.name),
-            handler_cname=(cd.exec_cd and cd.exec_cd.get_handler_cname()) or "NULL",
-            help_cname_ref=(cd.exec_cd and "&" + cd.exec_cd.get_help_cname()) or "NULL",
-            group_help_cname=cd.get_help_cname(),
-        )
-        subcommands = (
-            cd.exec_cd and cd.subcommands and cd.subcommands[1:]
-        ) or cd.subcommands
-        formatted_string += "\n".join([createcd(child) for child in subcommands or []])
+        formatted_string = createcd_typegroup(cd)
     else:
         raise Exception("Not handled cd type")
 

--- a/librz/core/cmd_descs/cmd_descs_generate.py
+++ b/librz/core/cmd_descs/cmd_descs_generate.py
@@ -819,10 +819,12 @@ cf_text = CMDDESCS_C_TEMPLATE.format(
     helps="\n".join(helps),
     init_code="\n".join(init_code),
 )
-with open(os.path.join(args.output_dir, "cmd_descs.c"), "w") as f:
+with open(os.path.join(args.output_dir, "cmd_descs.c"), "w", encoding="utf8") as f:
     f.write(cf_text)
 if args.src_output_dir:
-    with open(os.path.join(args.src_output_dir, "cmd_descs.c"), "w") as f:
+    with open(
+        os.path.join(args.src_output_dir, "cmd_descs.c"), "w", encoding="utf8"
+    ) as f:
         f.write(cf_text)
 
 handlers_decls = filter(
@@ -833,8 +835,10 @@ handlers_decls = filter(
 hf_text = CMDDESCS_H_TEMPLATE.format(
     handlers_declarations="\n".join([handler2decl(t, h) for t, h in handlers_decls]),
 )
-with open(os.path.join(args.output_dir, "cmd_descs.h"), "w") as f:
+with open(os.path.join(args.output_dir, "cmd_descs.h"), "w", encoding="utf8") as f:
     f.write(hf_text)
 if args.src_output_dir:
-    with open(os.path.join(args.src_output_dir, "cmd_descs.h"), "w") as f:
+    with open(
+        os.path.join(args.src_output_dir, "cmd_descs.h"), "w", encoding="utf8"
+    ) as f:
         f.write(hf_text)

--- a/librz/include/rz_cmd.h
+++ b/librz/include/rz_cmd.h
@@ -434,12 +434,14 @@ typedef struct rz_cmd_desc_t {
 		struct {
 			RzCmdArgvModesCb cb;
 			int modes; ///< A combination of RzOutputMode values
+			RzOutputMode default_mode; ///< Make one of the modes the default one, used even when the special suffix is not specified.
 			int min_argc;
 			int max_argc;
 		} argv_modes_data;
 		struct {
 			RzCmdArgvStateCb cb;
 			int modes; ///< A combination of RzOutputMode values
+			RzOutputMode default_mode; ///< Make one of the modes the default one, used even when the special suffix is not specified.
 			int min_argc;
 			int max_argc;
 		} argv_state_data;
@@ -536,6 +538,7 @@ RZ_API RzCmdDesc *rz_cmd_desc_oldinput_new(RzCmd *cmd, RzCmdDesc *parent, const 
 RZ_API RzCmdDesc *rz_cmd_desc_fake_new(RzCmd *cmd, RzCmdDesc *parent, const char *name, const RzCmdDescHelp *help);
 RZ_API RzCmdDesc *rz_cmd_desc_parent(RzCmdDesc *cd);
 RZ_API RzCmdDesc *rz_cmd_desc_get_exec(RzCmdDesc *cd);
+RZ_API bool rz_cmd_desc_set_default_mode(RzCmdDesc *cd, RzOutputMode mode);
 RZ_API bool rz_cmd_desc_has_handler(const RzCmdDesc *cd);
 RZ_API bool rz_cmd_desc_remove(RzCmd *cmd, RzCmdDesc *cd);
 RZ_API void rz_cmd_foreach_cmdname(RzCmd *cmd, RzCmdDesc *begin, RzCmdForeachNameCb cb, void *user);

--- a/sys/check_meson_subproject.py
+++ b/sys/check_meson_subproject.py
@@ -18,7 +18,7 @@ meson_root = os.environ["MESON_SOURCE_ROOT"]
 subproject_filename = os.path.join(meson_root, "subprojects", subproject + ".wrap")
 
 try:
-    with open(subproject_filename, "r") as f:
+    with open(subproject_filename, "r", encoding="utf8") as f:
 
         is_wrap_git = False
         revision = None
@@ -43,7 +43,9 @@ try:
             subproject_dir = os.path.join(meson_root, "subprojects", directory)
             subproject_git_dir = os.path.join(subproject_dir, ".git")
             if os.path.isdir(subproject_dir) and os.path.isdir(subproject_git_dir):
-                with open(os.path.join(subproject_git_dir, "HEAD"), "r") as f:
+                with open(
+                    os.path.join(subproject_git_dir, "HEAD"), "r", encoding="utf8"
+                ) as f:
                     head = f.read().strip()
                 if head != revision:
                     sys.exit(1)

--- a/sys/create_tags_rz.py
+++ b/sys/create_tags_rz.py
@@ -9,6 +9,6 @@ import os
 from sys import argv
 
 for fname in argv[1:]:
-    with open(fname) as f:
+    with open(fname, encoding="utf8") as f:
         text = " ".join(f.read().splitlines())
     print("ft %s %s" % (os.path.basename(fname), text))

--- a/sys/syscall_preprocessing.py
+++ b/sys/syscall_preprocessing.py
@@ -8,8 +8,8 @@
 import re
 import sys
 
-with open(sys.argv[1]) as inf:
-    with open(sys.argv[2], "w") as outf:
+with open(sys.argv[1], encoding="utf8") as inf:
+    with open(sys.argv[2], "w", encoding="utf8") as outf:
         for line in inf:
             if not line.startswith("_") and "=" in line:
                 arr = re.split("=|,", line)

--- a/sys/version.py
+++ b/sys/version.py
@@ -12,7 +12,7 @@ meson_file = "meson.build"
 if len(sys.argv) > 1:
     meson_file = os.path.join(sys.argv[1], meson_file)
 
-with open(meson_file, "r") as f:
+with open(meson_file, "r", encoding="utf8") as f:
     # Read only first 10 lines of the meson file, looking for 'version: ' string
     for i in range(10):
         fields = [x.strip() for x in f.readline().strip().split(":")]

--- a/test/unit/legacy_unit/syscall/openbsd-gen.py
+++ b/test/unit/legacy_unit/syscall/openbsd-gen.py
@@ -5,7 +5,7 @@
 import copy
 import sys
 
-with open("/usr/include/sys/syscall.h", "r") as f:
+with open("/usr/include/sys/syscall.h", "r", encoding="utf8") as f:
     rec = {"name": None, "args": None}
     recs = {}
 
@@ -37,7 +37,7 @@ with open("/usr/include/sys/syscall.h", "r") as f:
 
             recs[int(callnum)] = copy.copy(rec)
             rec = {"name": None, "args": None}
-with open("openbsd.c", "w") as out:
+with open("openbsd.c", "w", encoding="utf8") as out:
     out.write('#include "r_syscall.h"\n\n/* syscall-openbsd */\n')
     out.write("RSyscallItem syscalls_openbsd_x86[] = {\n")
 

--- a/test/unit/test_cmd.c
+++ b/test/unit/test_cmd.c
@@ -1165,6 +1165,53 @@ bool test_state_output_concat_json(void) {
 	mu_end;
 }
 
+static RzCmdStatus x_default_mode_handler(RzCore *core, int argc, const char **argv, RzCmdStateOutput *state) {
+	mu_assert_eq(state->mode, RZ_OUTPUT_MODE_JSON, "x handler should be called with JSON mode");
+	mu_assert_notnull(state->d.pj, "x handler should be called with pj nonnull");
+	return RZ_CMD_STATUS_OK;
+}
+
+bool test_default_mode(void) {
+	RzCmdDescArg z_args[] = { { 0 } };
+	RzCmdDescHelp z_help = { 0 };
+	z_help.summary = "z summary";
+	z_help.args = z_args;
+
+	RzCmdDescArg x_args[] = { { 0 } };
+	RzCmdDescHelp x_help = { 0 };
+	x_help.summary = "x summary";
+	x_help.args = x_args;
+
+	RzCmd *cmd = rz_cmd_new(false);
+	RzCmdDesc *root = rz_cmd_get_root(cmd);
+	RzCmdDesc *z_cd = rz_cmd_desc_argv_modes_new(cmd, root, "z", RZ_OUTPUT_MODE_STANDARD | RZ_OUTPUT_MODE_JSON | RZ_OUTPUT_MODE_QUIET | RZ_OUTPUT_MODE_LONG_JSON, z_modes_handler, &z_help);
+	RzCmdDesc *x_cd = rz_cmd_desc_argv_state_new(cmd, root, "x", RZ_OUTPUT_MODE_JSON | RZ_OUTPUT_MODE_QUIET | RZ_OUTPUT_MODE_LONG_JSON, x_default_mode_handler, &x_help);
+
+	bool changed = rz_cmd_desc_set_default_mode(z_cd, RZ_OUTPUT_MODE_JSON);
+	mu_assert_false(changed, "default_mode was not changed because _STANDARD is already defined for z_cd");
+	mu_assert_eq(z_cd->d.argv_modes_data.default_mode, RZ_OUTPUT_MODE_STANDARD, "default_mode is still _STANDARD for z_cd");
+
+	changed = rz_cmd_desc_set_default_mode(x_cd, RZ_OUTPUT_MODE_JSON);
+	mu_assert_true(changed, "default_mode was changed for x_cd");
+	mu_assert_eq(x_cd->d.argv_state_data.default_mode, RZ_OUTPUT_MODE_JSON, "default_mode is now _JSON for x_cd");
+
+	RzCmdParsedArgs *pa = rz_cmd_parsed_args_new("x", 0, NULL);
+	RzCmdStatus status = rz_cmd_call_parsed_args(cmd, pa);
+	mu_assert_eq(status, RZ_CMD_STATUS_OK, "x handler should be correctly executed");
+	rz_cmd_parsed_args_free(pa);
+
+	pa = rz_cmd_parsed_args_new("x?", 0, NULL);
+	char *h = rz_cmd_get_help(cmd, pa, false);
+	mu_assert_strcontains(h, "x[jqJ]   # x summary (JSON mode)", "x help should contain the default=json mode");
+	mu_assert_strcontains(h, "xj      # x summary (JSON mode)", "x help should contain the json mode");
+	mu_assert_strcontains(h, "xq      # x summary (quiet mode)", "x help should contain the quiet mode");
+	free(h);
+	rz_cmd_parsed_args_free(pa);
+
+	rz_cmd_free(cmd);
+	mu_end;
+}
+
 int all_tests() {
 	mu_run_test(test_parsed_args_noargs);
 	mu_run_test(test_parsed_args_onearg);
@@ -1201,6 +1248,7 @@ int all_tests() {
 	mu_run_test(test_state_output_concat_table);
 	mu_run_test(test_state_output_concat_mix);
 	mu_run_test(test_state_output_concat_json);
+	mu_run_test(test_default_mode);
 	return tests_passed != tests_run;
 }
 

